oai_agents/agents/rl.py

```diff
@@ -285,7 +289,7 @@ def log_details(self, experiment_name, total_train_timesteps):
         print("Final sparse reward ratio: ", self.args.final_sparse_r_ratio)
 
 
-    def train_agents(self, total_train_timesteps, tag, exp_name=None):
+    def train_agents(self, total_train_timesteps, tag, exp_name=None, resume_ck_list=None):
```

Comment:
People may assume that setting the tag here will make us to save a trained agent with this tag, but this is a false assumption. We don't acutally use the input "tag" in this function. 

We need to remove tag from the inputs.

Suggested Changes:
1. def train_agents(self, total_train_timesteps, exp_name=None, resume_ck_list=None):
2. Make sure to also change lines, which use this train_agents()

Reply:
Tag was not added in Naren's PR, the only thing he added is "resume_ck_list". you can see that by comparing the red and green lines. Moreover, tag is used in this function to save the agents. This is a feature I added when I merged the remove_aamas PR.

Reply:
to be more precise, tag is passed in train_helper.py so we can specify which agent we want to load during the multi step training process

Reply:
@avv-va Could you help me to point out which line we used? I read this function carefully couple of times, but cannot figure out where we use the tag, which is the input of train_agent(). Thanks!

Reply:
yes! it's here: 
https://github.com/HIRO-group/multiHRI/blob/04896104bdf58e5499dd72692f2531e2181a57db/oai_agents/agents/rl.py#L343

Reply:
oh shit you are right 🤦‍♀️ thanks for noticing this 
@narendasan can you rename the `tag` in the input to `tag_for_returning_agent` for now until we have a better name for it?
so basically, changed these two lines here:
https://github.com/HIRO-group/multiHRI/blob/04896104bdf58e5499dd72692f2531e2181a57db/oai_agents/agents/rl.py#L288
And
https://github.com/HIRO-group/multiHRI/blob/04896104bdf58e5499dd72692f2531e2181a57db/oai_agents/agents/rl.py#L343

and use 'tag_for_returning_agent' instead of 'tag'?

Reply:
When self.checkpoint_rate is None, we did use it. However, I suppose it is always not None in our current code. 

Additionally, if we really need to let user to decide which tag, we need to let them use different name. 
For example, in line 298 and line 334, we don't write to tag. Instead, we write the outputs of self.save_agents to ck_tag.

Reply:
Hi @narendasan,

I agree with @avv-va's suggestion. However, if you prefer a simpler solution, you could rename the tag in line 298 and line 334 instead of renaming it in train_agents(), which would require updating multiple places where train_agents() is called.

I'm comfortable with either option and will leave the decision to you.

Thank you!

Reply:
The purpose of the tag that is used in the input of this function is not for saving but rather retrieving a specific tag of that agent during the multi step training. I agree that we need a better name instead of just 'tag', so I suggested 'tag_for_returning_agent' above.

Reply:
I agree with you. Let's go for this option: rename 'tag' to 'tag_for_returning_agent' in train_agent and line343, like @avv-va  suggested! 

@narendasan, sorry for the confusion. If you need any clarification, feel free to let us know.